x/concentrated-liquidity/fees_test.go

```diff
@@ -10,61 +10,134 @@ import (
 	"github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity/types"
 )
 
+// fields used to identify a fee position.
+type positionIdentifiers struct {
+	poolId    uint64
+	owner     sdk.AccAddress
+	lowerTick int64
+	upperTick int64
+}
+
 var (
 	oneEth = sdk.NewDecCoin(ETH, sdk.OneInt())
 )
 
 func (s *KeeperTestSuite) TestInitializeFeeAccumulatorPosition() {
-	defaultPoolId := uint64(1)
-	defaultLiquidityDelta := sdk.MustNewDecFromStr("10.0")
+	// Setup is done once so that we test
+	// the relationship between test cases.
+	// For example, that positions with non-zero liquidity
+	// cannot be overriden.
+	s.SetupTest()
+
+	var (
+		defaultPoolId         = uint64(1)
+		defaultLiquidityDelta = sdk.MustNewDecFromStr("10.0")
+		defaultPositionId     = positionIdentifiers{
+			defaultPoolId,
+			s.TestAccs[0],
+			DefaultLowerTick,
+			DefaultUpperTick,
+		}
+	)
+
+	withOwner := func(posId positionIdentifiers, owner sdk.AccAddress) positionIdentifiers {
+		posId.owner = owner
+		return posId
+	}
+
+	withUpperTick := func(posId positionIdentifiers, upperTick int64) positionIdentifiers {
+		posId.upperTick = upperTick
+		return posId
+	}
+
+	withLowerTick := func(posId positionIdentifiers, lowerTick int64) positionIdentifiers {
+		posId.lowerTick = lowerTick
+		return posId
+	}
+
+	clKeeper := s.App.ConcentratedLiquidityKeeper
+
+	err := clKeeper.CreateFeeAccumulator(s.Ctx, defaultPoolId)
+	s.Require().NoError(err)
+
 	type initFeeAccumTest struct {
-		setPoolAccumulator  bool
-		setExistingPosition bool
-		expectedPass        bool
+		name           string
+		liquidityDelta sdk.Dec
+		positionId     positionIdentifiers
+
+		expectedPass bool
 	}
-	tests := map[string]initFeeAccumTest{
-		"existing accumulator, new position": {
-			setPoolAccumulator:  true,
-			setExistingPosition: false,
-			expectedPass:        true,
+	tests := []initFeeAccumTest{
+		{
+			name:           "new zero liqudidity position",
+			liquidityDelta: sdk.ZeroDec(),
+			// Gets overwritten by the next test case
```

Comment:
nit: what do you think of making all test cases independent of each other and then resetting the testing environment every iteration of a test? IMO making an independent test case / environment for each table tests makes a safer / clearer environment to test

Reply:
The thing is, here specifically it is important to test the relationship between test cases. I've explored your suggestion and found it hard to implement without much code duplication. As a result, I concluded that it is better to have dependent test cases here.

For example, we must make sure that if we already created a position and it had liquidity, that position would not get overwritten. We also need to make sure that positions with different owners or ticks are completely distinct positions. By having the test cases share context, we can easily test this

Reply:
yeah makes sense, I also thought that we could be having the same pain point as this test somewhat has the property of story-driven tests, lets leave this as is !